src/Linux/Setup/rtvs-install

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env bash
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+USE_DPKG=false
+DEBIAN_DEPENDENCIES="libexplain51 libzip4 libc6 lshw ssl-cert locales"
+GETOPT_BIN=$(type getopt 2>/dev/null)
+DOTNET_SOURCE_ADDED=false
+DOTNET_INSTALLED=false
+DOTNET_BIN=$(type dotnet 2>/dev/null)
+
+SILENT=false
+LOCALE_GEN=true
+
+function error {
+  echo -e "$*" >&2
+}
+
+function message {
+  if [ ${SILENT} == true ]; then
+    return 0
+  fi
+  printf "$1"
+}
+
+function print_help {
+cat <<EOF
+  Usage: rtvs-install [OPTIONS]
+
+    -s, --silent                 perform a silent, unattended install
+    -l, --no-locale-gen          does not call locale-gen during install
+    -h, --help                   print this help text
+
+EOF
+}
+
+function parse_options {
+  local TEMP=$(getopt -o -lsuh --long no-locale-gen,silent,help -n "${PROG_NAME}" -- "$@")
+
+  if [ $? != 0 ]; then
+    echo "Option string malformed" >&2
+    exit 1
+  fi  
+
+  eval set -- "${TEMP}"
+
+  while true; do
+    case "$1" in
+      -s|--silent)                  SILENT=true               ; shift ;;
+      -l|--no-locale-gen)           LOCALE_GEN=false          ; shift ;;
+      -h|--help)                    print_help                ; exit  ;;
+      --) shift; break ;;
+      *) echo "Internal error while parsing options $*" >&2   ; exit 1 ;;
+    esac
+  done
+
+}
+
+function check_root {
+  if [[ "$(id -u)" != "0" ]]; then
+      error  "This script must be run with sudo or as root."
+      exit 1
+  fi
+}
+
+function check_getopt {
+  if [[ -z ${GETOPT_BIN} ]]; then
+    echo ""
+    echo "This script requires GNU getopt to run. Install this utility and rerun this script."
```

Comment:
`getopt` is a standard POSIX utility, so it's always going to be there. But it might not be GNU getopt specifically. The use of long options above is a GNU extension, so this does indeed need GNU getopt. 

I think a better way to check for this is to invoke something like `getopt -o -x --long x`, and see if it fails or not (from error code).

Reply:
(Or, alternatively, drop the long options, and use the strict POSIX subset of `getopt`)